src/Components/Components/src/PersistentState/ComponentStatePersistenceManager.cs

```diff
@@ -59,6 +78,9 @@ public Task PersistStateAsync(IPersistentComponentStateStore store, Renderer ren
 
         async Task PauseAndPersistState()
         {
+            // Ensure that we register the services before we start persisting the state.
+            _servicesRegistry?.RegisterForPersistence(State);
```

Comment:
The naming of this makes me suspicious. It sounds like it would create duplicate callbacks every time the circuit is paused and persisted.

Do we somehow know this can only get called once? Or do we know that `RegisterForPersistence` is idempotent? Is there possibly some change of naming that would make this all clearer?

Reply:
This only gets called once at the end of the lifetime of the host (during prerendering in this case, in the future when we "hibernate" the circuit.

If in the future there's a situation where this gets called more than once, we can adjust